src/main/java/dev/nolij/zson/Zson.java

```diff
@@ -127,6 +146,89 @@ public static String escape(String string, char escapeQuotes) {
 		return result.toString();
 	}
 
+	@NotNull
+	@Contract("_ -> new")
+	public static Map<String, ZsonValue> obj2Map(@NotNull Object object) {
+		Map<String, ZsonValue> map = Zson.object();
+		for (Field field : object.getClass().getDeclaredFields()) {
+			if(!shouldInclude(field, true)) continue;
+			Comment comment = field.getAnnotation(Comment.class);
+			String commentValue = comment == null ? null : comment.value();
+			try {
+				map.put(field.getName(), new ZsonValue(commentValue, field.get(object)));
+			} catch (IllegalAccessException e) {
+				throw new RuntimeException("Failed to get field " + field.getName(), e);
+			}
+		}
+		return map;
+	}
+
+	@NotNull
+	@Contract("_ , _ -> new")
+	public static <T> T map2Obj(@NotNull Map<String, ZsonValue> map, @NotNull Class<T> type) {
+		try {
+			T object = type.getDeclaredConstructor().newInstance();
+			for (Field field : type.getDeclaredFields()) {
+				if(!shouldInclude(field, false)) continue;
+				setField(field, object, map.get(field.getName()).value);
+			}
+			return object;
+		} catch (Exception e) {
+			throw new RuntimeException("Failed to create object of type " + type.getSimpleName(), e);
+		}
+	}
+
+	private static boolean shouldInclude(Field field, boolean includeStatic) {
```

Comment:
Pseudo-code replacement: 
```java
if (isExcluded)
    return false;

if (isStatic)
    return isIncluded;

return isPublic || isIncluded;
```

Reply:
The new implementation still doesn't fully match this, and isn't as easy to follow. Please at least split the condition in to separate lines and add comments next to the conditions explaining them for readability.